design-proposals/backend-storage-block-migration.md

```diff
@@ -0,0 +1,62 @@
+# Overview
+This proposal is about supporting the migration of all VMs that use a persistent feature, such as persistent vTPM.
+
+To make this document as simple as possible, let's consider that storage in Kubernetes can either be filesystem (FS) or
+block mode, and either ReadWriteOnce (RWO) or ReadWriteMany (RWX) access mode, with storage classes supporting any
+number of combinations of both.
+
+As of KubeVirt 1.3, Here is how each of the 4 combinations is supported by the backend-storage for persistent VM state:
+- RWO FS: supported by backend-storage but makes VMs non-migratable
+- RWX FS: preferred combination, supported by backend-storage and compatible with VM migrations
+- RWO block: not supported yet, but PRs exist to implement it in a non-migratable way 
+- RWX block: same as above, with migratability functional but considered unsafe (as it concurrently mounts an ext3 partition)
+
+This is the best we can do with the current implementation, which involves mounting the same partition on migration source and target.
+This proposal is about switching to a different model which would instead create a new backend-storage PVC on migration,
+allowing all 4 cases above to have proper migration support.
```

Comment:
The point of this proposal is to stop mounting the PVC on both source and target, to enable migratability of block-based backend-storage.

Reply:
I do not understand why we are interested with RWO. If we have only RWO, we cannot migrate the root disk. I don't understand why we conflate this proposal (and future tests) with this combination (migration+RWO).

Reply:
The backend-sotage may be set to a different storage class than the other disks.
What is being discussed here is that since the content of this volume will be copied, the access mode of such volume will become irrelevant and might as well be RWO.

Reply:
Thinking about RWO and discussing it makes this proposal more complex and slower to test. What is the benefit of it?

Reply:
From my side, we can ignore it. However, in many cases, this is the RWO is the default access mode. We don't have to test or support this.

Reply:
What about containerdisk VMs?  They can be live migrated, correct?  Can we use emptydir volumes for backend storage in that case?

Reply:
Using emptydir for backend storage is effectively what happens for non-persistent TPM/EFI.
If we want persistence, we need a PVC.